### PR TITLE
CSV export of search results

### PR DESCRIPTION
**Pull request by ketilaa**
- text/csv offered as content-negotiation rather than separate endpoint
- frontend should limit size of export to 100 when using text/csv (start=0, results=100)
- generated CSV now optimised for use in MS Excel on Windows

More refactoring could be done for tests in this repository, but I had to set a line to deliver this new feature.

### CODE REVIEW

File: search-commons/src/test/java/no/unit/nva/search/ExportSearchResourcesTest.java [hunk 1]
```diff
@@ -27,7 +27,8 @@ void shouldAllowCreationOfCsv() throws IOException {
         var searchResponse = SearchResponseDto.fromSearchResponse(getSearchResponseFromJson(json), randomUri());
 
         var value = ExportSearchResources.exportSearchResults(searchResponse);
-        var actual = CsvUtil.toExportCsv(value);
+        // skipping UTF-8 BOM for some reason is needed
+        var actual = CsvUtil.toExportCsv(value.substring(1));
```
Comment (brinxmat) [inline search-commons/src/test/java/no/unit/nva/search/ExportSearchResourcesTest.java:31]: Can you extract this to a method called `removeUtfBom`to make this very clear that this is happening. Were we not in complete control, I'd even suggest ensuring that the BOM was a BOM, but we are, so don't

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java [hunk 2]
```diff
@@ -198,33 +205,157 @@ void shouldReturnBadGatewayResponseWhenNoResponseFromService() throws IOExceptio
...
+    }
+
+    @Test
+    void shouldReturnTextCsvIfAcceptTextCsvProvided() throws IOException {
+        prepareRestHighLevelClientOkResponse(List.of(csvWithFullDate(), csvWithYearOnly()));
+        handler.handleRequest(getRequestInputStreamAccepting("text/csv"), outputStream, mock(Context.class));
+
+        GatewayResponse<String> gatewayResponse = GatewayResponse.fromOutputStream(outputStream, String.class);
+        assertThat(gatewayResponse.getHeaders().get("Content-Type"), is(equalTo("text/csv; charset=utf-8")));
+    }
```
Comment (brinxmat) [inline search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:242]: ParameterisedTest? Also text/html…

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java [hunk 3]
```diff
@@ -41,11 +46,32 @@ public SearchResourcesApiHandler(Environment environment, SearchClient openSearc
...
+        var searchResponse = openSearchClient.searchWithSearchDocumentQuery(query,
+                                                                          OPENSEARCH_ENDPOINT_INDEX);
+        return createResponse(requestInfo, searchResponse);
+    }
+
+    private String createResponse(RequestInfo requestInfo, SearchResponseDto searchResponse)
+        throws UnsupportedAcceptHeaderException {
+        var contentType = getDefaultResponseContentTypeHeaderValue(requestInfo);
+
+        if ("text".equalsIgnoreCase(contentType.type())) {
```
[the remark this answers is not in the document]
Comment (ketilaa) [reply search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java]: I am not aware of any requirements for users of web browsers for this endpoint. Should we care?

File: search-commons/src/test/java/no/unit/nva/search/ExportSearchResourcesTest.java (hunk 1 above)
Comment (ketilaa) [reply search-commons/src/test/java/no/unit/nva/search/ExportSearchResourcesTest.java:31]: Fixed

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java (hunk 2 above)
Comment (ketilaa) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java]: Added parameterised tests. What is requirement for `text/html`?

**Review by brinxmat — APPROVED** [review]

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java (hunk 2 above)
Comment (brinxmat) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java]: Not sure…I think I might just expect json…

File: search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (brinxmat) [reply search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java]: Tror at jeg ikke forventer CSV i nettleser…

**Pull request merged**